cpp/tests/c_api/mg_pagerank_test.c

```diff
@@ -0,0 +1,257 @@
+int test_pagerank_4(const cugraph_resource_handle_t* handle)
+{
+  size_t num_edges    = 3;
+  size_t num_vertices = 4;
+
+  vertex_t h_src[]    = {0, 1, 2};
+  vertex_t h_dst[]    = {1, 2, 3};
+  weight_t h_wgt[]    = {1.f, 1.f, 1.f};
+  weight_t h_result[] = {
+    0.11615584790706635f, 0.21488840878009796f, 0.29881080985069275f, 0.37014490365982056f};
+
+  double alpha          = 0.85;
+  double epsilon        = 1.0e-6;
+  size_t max_iterations = 500;
+
+  return generic_pagerank_test(handle,
+                               h_src,
+                               h_dst,
+                               h_wgt,
+                               h_result,
+                               num_vertices,
+                               num_edges,
+                               FALSE,
+                               alpha,
+                               epsilon,
+                               max_iterations);
+}
+
+int test_pagerank_4_with_transpose(const cugraph_resource_handle_t* handle)
+{
+  size_t num_edges    = 3;
+  size_t num_vertices = 4;
+
+  vertex_t h_src[]    = {0, 1, 2};
+  vertex_t h_dst[]    = {1, 2, 3};
+  weight_t h_wgt[]    = {1.f, 1.f, 1.f};
+  weight_t h_result[] = {
+    0.11615584790706635f, 0.21488840878009796f, 0.29881080985069275f, 0.37014490365982056f};
+
+  double alpha          = 0.85;
+  double epsilon        = 1.0e-6;
+  size_t max_iterations = 500;
+
+  return generic_pagerank_test(handle,
+                               h_src,
+                               h_dst,
+                               h_wgt,
+                               h_result,
+                               num_vertices,
+                               num_edges,
+                               TRUE,
+                               alpha,
+                               epsilon,
+                               max_iterations);
+}
+
+/******************************************************************************/
+
+int main(int argc, char** argv)
+{
+  // Set up MPI:
+  int comm_rank;
+  int comm_size;
+  int num_gpus_per_node;
+  cudaError_t status;
+  int mpi_status;
+  int result                        = 0;
+  cugraph_resource_handle_t* handle = NULL;
+  cugraph_error_t* ret_error;
+  cugraph_error_code_t ret_code = CUGRAPH_SUCCESS;
+  int prows                     = 1;
+
+  C_MPI_TRY(MPI_Init(&argc, &argv));
+  C_MPI_TRY(MPI_Comm_rank(MPI_COMM_WORLD, &comm_rank));
+  C_MPI_TRY(MPI_Comm_size(MPI_COMM_WORLD, &comm_size));
+  C_CUDA_TRY(cudaGetDeviceCount(&num_gpus_per_node));
+  C_CUDA_TRY(cudaSetDevice(comm_rank % num_gpus_per_node));
+
+#if 0
+  // TODO:  Need something a bit more sophisticated for bigger systems
+  prows = (int)sqrt((double)comm_size);
+  while (comm_size % prows != 0) {
+    --prows;
+  }
+
+  ret_code = cugraph_resource_handle_init_comms(handle, prows, &ret_error);
+  TEST_ASSERT(result, ret_code == CUGRAPH_SUCCESS, "handle create failed.");
+  TEST_ASSERT(result, ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
```

Comment:
Same here. We may consider adding a utility function to set prows from p in our C++ main codebase and call that function in  MG C++ tests, MG C tests, and python tests.

Reply:
Umm... not sure how all this is related to computing p_rows from p, but I get this is not directly relevant to this PR. We should discuss how to properly compute p_rows (& p_cols) from p, but I agree that this should be a separate discussion.